README.md

```diff
@@ -1,45 +1,224 @@
+      namespace: default
+    spec:
+      randomkey: samplevalue
+  crds:
+    ref: sha256:71cf4f1fee1a2f51296cc805aa9b24bc14fd5c2b4aee1e24aadc2996b067bb3d
+    name: kyma-project.io/module/example-module-name
+    repo: kcp-registry.localhost:8888/component-descriptors
+    type: oci-ref
+  config:
+    ref: sha256:61be4f1fee1a2f51296cc805aa9b24bc14fd5c2b4aee1e24aadc2996b067ccec
+    name: kyma-project.io/module/example-module-name
+    repo: kcp-registry.localhost:8888/component-descriptors
+    type: oci-ref
+  installs:
+    - source:
+        name: kyma-project.io/module/example-module-name
+        repo: kcp-registry.localhost:8888/component-descriptors
+        ref: sha256:c64f0580a74259712f24243528881a76b5e1c9cd254fa58197de93a6347f99b9
+        type: oci-ref
+      name: redis
+    - source:
+        chartName: mysql
+        url: https://charts.bitnami.com/bitnami
+        type: helm-chart
+      name: bitnami
 ```
-    manifestOperations := manifest.NewOperations(logger, r.RestConfig, cli.New())
-	if create {
-		if err := manifestOperations.Install("chart/path", releaseName, "repoName/chartName", repoName, url, args); err != nil {
-			return err
-		}
-	} else {
-		if err := manifestOperations.Uninstall("chart/path", "repoName/chartName", releaseName, args); err != nil {
-			return err
-		}
-	}
-  ```
-- `chartPath` takes priority over the **url** parameter. If the chart should be downloaded from `url` pass `chartPath` as empty string
-- `args` only supports `--set` and [action flags](https://github.com/helm/helm/blob/v3.9.0/pkg/action/install.go#L66), with variable names and values, comma-seperated. Example:
+
+## Manifest library
+
+The operator uses the [manifest library](https://pkg.go.dev/github.com/kyma-project/module-manager/operator/pkg/manifest) to process deployments on clusters.
+
+It supports helm chart installations from two sources: **_helm repositories_** and **_local paths_**. Additionally, it helps to process additional local installations of CRs, CRDs and custom state checks.
+
+>Note: We plan to offer [Kustomize installation support](https://github.com/kyma-project/module-manager/issues/124), along with the existing helm installation soon. 
+
+The library could be used to simply process deployments on target clusters or use within your own operator to carry deployment operations. 
+E.g. [template-operator](https://github.com/kyma-project/lifecycle-manager/tree/main/samples/template-operator) uses this library (via the [declarative](operator/pkg/declarative) library) to perform necessary operations on target clusters during reconciliations.
+To get started, simply import package `github.com/kyma-project/module-manager/operator/pkg/manifest` to include the main functionality provided by the library to process helm charts, coupled with additional state handling.
+For more options and information refer to the [InstallInfo](operator/pkg/manifest/operations.go) type definition.
+
+### Sample usage
+
+```go
```

Comment:
Since the sample it pretty long, maybe put it inside a collapsable block - like here: https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

Reply:
added this both for sample Manifest resource and sample library usage